src/libs/OptionsListUtils.js

```diff
@@ -58,12 +58,11 @@ Onyx.connect({
 
 const iouReports = {};
 Onyx.connect({
-    key: ONYXKEYS.COLLECTION.REPORT_IOUS,
+    key: ONYXKEYS.COLLECTION.REPORT,
     callback: (iouReport, key) => {
-        if (!iouReport || !key || !iouReport.ownerEmail) {
+        if (!iouReport || !key || !iouReport.ownerEmail || !ReportUtils.isIOUReport(iouReport)) {
```

Comment:
Not sure if it happens, but what if we are trying to remove an IOU report (`iouReport` is `null` and `key` is not falsey)?, shouldn't we do `delete iouReports[key]` in this case?

Reply:
In current behavior, I just see it populated as `null` in the list, which ends up with the right behavior 🤷